src/main/java/ch/njol/skript/entity/BoatChestData.java

```diff
@@ -34,18 +34,31 @@
 public class BoatChestData extends EntityData<ChestBoat> {
 
 	static {
+		// This ensures all boats are registered
+		// As well as in the correct order via 'ordinal'
+		int boatsSize = Boat.Type.values().length;
+		String[] patterns = new String[boatsSize + 2];
+		patterns[0] = "chest boat";
+		patterns[1] = "any chest boat";
+		for (Boat.Type boat : Boat.Type.values()) {
+			String boatName = boat.toString().replace("_", " ").toLowerCase();
+			if (boatName.equals("bamboo"))
+				boatName = boatName + " chest raft";
+			else
+				boatName = boatName + " chest boat";
+			patterns[boat.ordinal() + 2] = boatName;
+		}
+
 		if (Skript.classExists("org.bukkit.entity.ChestBoat")) {
-			EntityData.register(BoatChestData.class, "chest boat", ChestBoat.class, 0,
-				"chest boat", "any chest boat", "oak chest boat", "spruce chest boat", "birch chest boat",
-				"jungle chest boat", "acacia chest boat", "dark oak chest boat");
+			EntityData.register(BoatChestData.class, "chest boat", ChestBoat.class, 0, patterns);
 		}
 	}
 
 	public BoatChestData() {
 		this(0);
 	}
 
-	public BoatChestData(@Nullable TreeSpecies type) {
+	public BoatChestData(Boat.Type type) {
```

Comment:
Can still be nullable
```suggestion
	public BoatChestData(@Nullable Boat.Type type) {
```

Reply:
Not resolved